src/test/name_applications_tests.cpp

```diff
@@ -119,4 +119,14 @@ BOOST_AUTO_TEST_CASE( name_description )
     BOOST_CHECK(DescFromName(identityValid, NameNamespace::Identity) == "'id/wikileaks'");
 }
 
+BOOST_AUTO_TEST_CASE( valid_json )
+{
+    // JSON string from https://namebrow.se/name/d/testls/, from block 699282
+    BOOST_CHECK_EQUAL(IsValidJSON("{\"ip\":\"107.152.38.155\",\"map\":{\"*\":{\"tls\":[[2,1,0,\"MDkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDIgADvxHcjwDYMNfUSTtSIn3VbBC1sOzh/1Fv5T0UzEuLWIE=\"]]},\"sub1\":{\"map\":{\"sub2\":{\"map\":{\"sub3\":{\"ip\":\"107.152.38.155\"}}}}},\"_tor\":{\"txt\":\"dhflg7a7etr77hwt4eerwoovhg7b5bivt2jem4366dt4psgnl5diyiyd.onion\"}}}"), true);
+    
+    // Removed some characters from the previous string for an invalid JSON value.
+    BOOST_CHECK_EQUAL(IsValidJSON("{\"ip\":\"107.152.38.155\",\"map\":{\":[[2,1,0,\"MDkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDIgADvxHcjwDYMNfUSTtSIn3VbBC1sOzh/1Fv5T0UzEuLWIE=\"]]},\"sub1\":{\"map\":{\"sub2\":{\"map\":{\"sub3\":{\"ip\":\"107.152.38.155\"}}}}},\"_tor\":{\"txt\":\"dhflg7a7etr77hwt4eerwoovhg7b5bivt2jem4366dt4psgnl5diyiyd.onion\"}}}"), false);
+    
```

Comment:
The test here doesn't matter too much, since the main logic is from Univalue anyway (and so you are just testing whether or not you use Univalue correctly to parse JSON).

I do find it distracting that such a complex value is used, though.  For your purposes, you don't need a real value from Namecoin at all, and could use something very simple like `null`, `false`, `"foo"` or `{"bar": [1, 2, 3]}` as valid values, and `{"foo":` as invalid (or many other simple strings obviously).

Reply:
I prefer a JSON object rather than a non-object JSON value such as a string or boolean, but other than that, yes this is a reasonable request.

Reply:
Well, that depends on what the logic should be.  If any JSON value should be acceptable, then I think the test should probably explicitly run both object and non-object values through the function.  If we only want to accept JSON objects, then obviously the logic itself should be updated (we have exactly the same validation, although on the consensus level, in Xaya btw), and the test verify that, too.

Reply:
My rough vision is that a subsequent PR would add some validation on specific JSON fields, and that when we do that, restricting to a JSON object would make sense (at which point we'd add more tests accordingly). I don't see any reason to tie that to this PR (incrementalism is good), but I also don't see any reason to explicitly test behavior that's only temporary and which will be changed (and which no users are relying on).